lib/Doctrine/Validator/Notblank.php

```diff
@@ -41,6 +41,6 @@ class Doctrine_Validator_Notblank extends Doctrine_Validator_Driver
      */
     public function validate($value)
     {
-        return (trim($value) !== '' && $value !== null);
+        return (trim((string) $value) !== '' && $value !== null);
```

Comment:
You could either swap the two operands and remove the cast or remove the second condition.

Reply:
Thank you very much, it avoid a useless cast.